community-tutorials/ruby-on-rails-in-production/01-en.md

```diff
@@ -0,0 +1,269 @@
+```
+
+## Configuring NGINX & Passenger
+
+
+Passenger comes with its own compiled Ruby version by default, I don't recommend using this as it can be outdated and we installed rbenv in advance which solves a lot of problems for us.
+
+To adjust this we need to edit the following file:
+
+```sh
+sudo nano /etc/nginx/conf.d/mod-http-passenger.conf
+```
+
+There we change the line to the following:
+
+```diff
+- passenger_ruby /usr/bin/passenger_free_ruby;
++ passenger_ruby /home/rails_user/.rbenv/shims/ruby;
+```
+
+Once we have saved the file, we execute the following command to start NGINX:
+
+```sh
+sudo service nginx start
+```
+
+If we now access the IP or address of the server via Webbrowser and receive the message "Welcome to NGINX", we have configured NGINX correctly.
+
+
+Now we can create a new configuration file by creating the following file with the editor nano:
+
+```sh
+sudo nano /etc/nginx/sites-enabled/rails_app
+```
+
+We copy the following content into it:
+
+```
+server {
+  listen 80;
+  listen [::]:80;
+
+  server_name YOUR_DOMAIN;
+  root /var/www/rails_app/public;
+
+  passenger_enabled on;
+  passenger_app_env production;
+
+  location /cable {
+    passenger_app_group_name myapp_websocket;
+    passenger_force_max_concurrent_requests_per_process 0;
+  }
+
+  client_max_body_size 10m;
+
+  location ~ ^/(assets|packs) {
+    expires max;
+    gzip_static on;
+  }
+}
+```
+
+You can modify the following details according to your needs:
+
+* server_name
+  * Instead of the `YOUR_DOMAIN`, you can enter your domain here on which your Rails app should run.
+* root
+  * `/var/www/rails_app` is the location of our Rails app. You can customise this, the important thing is that the path always ends with `/public`.
+* passenger_app_env
+  * If you prefer to run your Rails app for development in the development environment, you can specify `development` here instead of `production`.
+* location /cable
+  * If you are sure that your app will not use an action cable for websockets, you can remove this block from the file.
+* client_max_body_size
+  * If you want to allow your users to upload files to your Rails app, you can set a maximum upload file size here. In the default setting, this is 10 MB.
+* location ~ ^/(assets|packs)
+  * The assets and packs directories contain javascript and css files from your Rails app. To make these files more performant, they are cached and zipped with the block. You can extend this with more directories.
+
+
+Once you have saved the file, we change to the directory `/var/www/`. Now you can copy your existing Rails app into the folder `rails_app`. Alternatively, you can use tools like Capistrano for deployment.
+
+To install Rails, use the gem install command provided by RubyGems:
+
+```sh
+gem install rails
+```
+
+If you want to start from scratch, you can create a new rails app here with the name `rails_app` and execute the following command:
+
+```sh
+rails new rails_app
```

Comment:
I can´t run this code with rails_user.

Reply:
I think you have not installed rbenv as rails_user. That is why these errors occur.

Reply:
I reran everything again and I had to change the permissions on `/var/www` to run this command without errors. 
Now, when I visit my server, I get the default nginx page again.

Reply:
Hmm strange, for me it works with the domain. Can you delete the default config again. Maybe you can also post your config. I have information from the following link. There it is also not documented differently.

https://www.phusionpassenger.com/library/deploy/nginx/deploy/ruby/

Reply:
I have now deleted the nginx defailt config file and now I´m getting the Rails error page. (I guess this is a success)

Reply:
Yes perfect. The error is displayed I think because the request is not routed to a controller.